### ISSUE TITLE
cannot run spyder in a conda virtual environment for  specific settings

### ISSUE THREAD

**Issue by isalirezag** [spyder-ide/spyder#7938]
im using ubuntu 16.04 and anaconda 3.
Python 3.6.6 64bits, Qt 5.9.6, PyQt5 5.9.2 on Linux 
I am using  Spyder 3.3.1 , 

I created an virtual environment via conda (conda create -n PyTorch_0.3
) and inside that i installed a pytorch version (pytorch 0.3) that i want for a specific project . I also have the newest version of pytorch (pytorch 0.4.1) outside of my virtual environment .

Now I activate my virtuan env and go there and run spyder in that terminal, but when i check the version of pytorch there it still shows that it is using the newest pytorch not the version that i have inside that virtual env.
Can you please help what should i do to use the version that i installed inside the virtual env?

in other words, when i check the version in the terminal it shows it okay:
![image](https://user-images.githubusercontent.com/12434910/45930490-fedb8400-bf2e-11e8-8a7b-b8dcbb9b02d0.png)

but when i open spyder:
![image](https://user-images.githubusercontent.com/12434910/45930523-57128600-bf2f-11e8-8b2b-1b4978ad0aa9.png)

and check the version it does not match:

![image](https://user-images.githubusercontent.com/12434910/45930513-3ea26b80-bf2f-11e8-8008-c53896592e6b.png)

Thanks

**Comment by isalirezag** [comment]
@ccordoba12 do you have any suggestion?

**Comment by ccordoba12** [comment]
You probably have other Python interpreter selected in

   Tools > Preferences > Python interpreter

That's why you're not getting the result you expect.

**Comment by isalirezag** [comment]
it is like this:
![image](https://user-images.githubusercontent.com/12434910/46216485-ae3ca000-c30d-11e8-96dc-7d829651403d.png)

should i change it to "USe the follwoing Python interpreter" "? what should i do in the path if i change to this option?

**Comment by ccordoba12** [comment]
Please read

https://github.com/spyder-ide/spyder/wiki/Working-with-packages-and-environments-in-Spyder

to see if you can find the solution there.

**Comment by isalirezag** [comment]
I read it.
I am doing pretty much as it said. the difference is that, for models like keras, we can install the version that we want when creating thevirtual environment but for pytorch we cannot, so i installed it separately inside the virtual environment, when check it in the virtual environment it works, but in spyder is not. i m not sure why it happens :/

**Comment by ccordoba12** [comment]
Please give us a step by description of how you're creating your virtualenv, so we can test the process in our side.

**Comment by isalirezag** [comment]
i mentioned the details on the first comments.
please let me know if it is not clear

**Comment by isalirezag** [comment]
I have that in my system, no virtual environment for that.
It has nothing to do with python i guess, it is PyTorch not Python.
The python is the same for both of them

**Comment by ccordoba12** [comment]
> I have Pytorch 0.4.1 in my main system

As I said, that's a bad idea. What happens if you remove that package and try again?

**Comment by ccordoba12** [comment]
Did you install Spyder in your virtualenv too? Because if not, you're starting your system Spyder.

**Comment by isalirezag** [comment]
no it was already installed via anaconda, do you think i should reintall it in the virtual env?

**Comment by ccordoba12** [comment]
Of course because if not the Spyder that starts with `spyder` in the command line is the one in your root environment (and that's the environment with PyTorch 0.4).

This is documented documented in the guide you said you already read:

https://github.com/spyder-ide/spyder/wiki/Working-with-packages-and-environments-in-Spyder#the-naive-approach

Please don't make us waste time for next time.
